File: frontend/src/types/capabilities.ts
import {
  CriteriaKey,
  FeedbackStatus,
  Provenance,
  SlideKey,
  Topic,
  TurnIntent,
} from './contracts';
import { Gap } from './profile';

export interface IdeaAnalysis {
  problem: string;
  target_customer: string;
  solution: string;
  pain_points: string[];
  assumptions: string[];
  risks: string[];
  missing_information: string[];
  first_question: {
    text: string;
    target_field: string;
    topic: Topic;
  };
}

export interface ClarificationQuestion {
  id: string;
  text: string;
  target_field: string;
  topic?: Topic;
  reasoning?: string;
}

export interface ProfilePatchOp {
  op: 'set' | 'add' | 'remove';
  path: string;
  value: unknown;
  reason: string;
}

export interface ProfilePatch {
  ops: ProfilePatchOp[];
}

export interface Claim {
  text: string;
  provenance: Provenance;
  evidence_ids?: string[];
  supporting_quote?: string | null;
}

export interface SlideContent {
  key: SlideKey;
  title: string;
  bullets: Array<Claim | string>;
  status: 'complete' | 'needs_input';
}

export interface Pitch {
  slides: SlideContent[];
}

export interface PitchCritiqueIssue {
  slide_key: SlideKey;
  severity: 'low' | 'med' | 'high';
  title: string;
  explanation: string;
  suggestion: string;
}

export interface PitchCritique {
  overall_score: number;
  issues: PitchCritiqueIssue[];
  investor_objections: string[];
  top_priorities: string[];
}

export interface ValueProposition {
  one_line: string;
  elevator: string;
  differentiation: string;
  advantage: string;
  target_segment: string;
  customer_problem_alignment: string;
  provenance: Record<string, Provenance>;
}

export interface CompetitorDetail {
  name: string;
  description: string;
  strengths: string[];
  weaknesses: string[];
  pricing: string;
  market_share: string;
  differentiator: string;
  provenance: Provenance;
  source_urls: string[];
}

export interface CompetitorPosition {
  name: string;
  x: number; // -10 to 10
  y: number; // -10 to 10
  is_self?: boolean;
}

export interface CompetitorAnalysis {
  direct_competitors: CompetitorDetail[];
  indirect_competitors: Array<{ name: string; description: string }>;
  matrix_axes: {
    x_axis: string;
    y_axis: string;
  };
  competitor_positions: CompetitorPosition[];
}

export interface MarketTier {
  value: number;
  formatted: string;
  description: string;
  calculation_steps: string[];
  provenance: Provenance;
  sources: string[];
}

export interface MarketAnalysis {
  tam: MarketTier;
  sam: MarketTier;
  som: MarketTier;
  cagr: number;
  growth_drivers: string[];
  market_risks: string[];
  assumptions: string[];
}

export interface RevenueStream {
  name: string;
  model_type: string;
  price_point: string;
  frequency: string;
  margin: string;
  calculated: boolean;
  provenance: Provenance;
}

export interface PricingTier {
  tier_name: string;
  price: string;
  billing: string;
  features: string[];
  target_audience: string;
}

export interface UnitEconomics {
  cac: number;
  ltv: number;
  payback_months: number;
  gross_margin: number;
  ltv_cac_ratio: number;
  provenance: Provenance;
  formulas: Record<string, string>;
}

export interface BusinessModelAnalysis {
  type: string;
  revenue_streams: RevenueStream[];
  pricing_tiers: PricingTier[];
  unit_economics: UnitEconomics;
  cost_structure: string[];
}

export interface InvestorQuestion {
  id: string;
  text: string;
  intent: TurnIntent;
  topic: Topic;
}

export interface AnswerEvaluation {
  scores: Record<CriteriaKey, number>;
  explanation: string;
  strengths: string[];
  weaknesses: string[];
  recommended_improvement: string;
  weakest_criterion: CriteriaKey;
  claims_made: string[];
  follow_up_question: string | null;
  new_gaps?: Gap[];
}

export interface ReadinessNarrative {
  overall_score: number;
  verdict: string;
  executive_summary: string;
  criteria_scores: Record<CriteriaKey, number>;
  topic_coverage: Record<Topic, number>;
  top_strengths: string[];
  critical_gaps: string[];
  action_items: string[];
  investor_verdict: string;
}

export interface FeedbackItemDraft {
  id: string;
  topic: Topic;
  title: string;
  description: string;
  source: 'critique' | 'investor' | 'peer';
  status: FeedbackStatus;
  suggested_patch: ProfilePatch | null;
}

File: backend/app/ai/schemas/capabilities.py
from typing import Any, Literal
from pydantic import BaseModel, Field

class IdeaAnalysis(BaseModel):
    problem: str
    target_customer: str
    solution: str
    pain_points: list[str]
    assumptions: list[str]
    risks: list[str]
    missing_information: list[str]
    first_question: dict[str, Any]

class ClarificationQuestion(BaseModel):
    id: str
    text: str
    target_field: str

class ProfilePatchOp(BaseModel):
    op: Literal["set", "add", "remove"]
    path: str
    value: Any
    reason: str

class ProfilePatch(BaseModel):
    ops: list[ProfilePatchOp]

class Claim(BaseModel):
    text: str
    provenance: Literal["source_backed", "founder_assumption", "calculated", "ai_analysis"]
    evidence_ids: list[str] = Field(default_factory=list)
    supporting_quote: str | None = None

class SlideContent(BaseModel):
    key: Literal[
        "problem", "solution", "product", "target_market", "market_opportunity",
        "competition", "advantage", "business_model", "gtm", "traction", "team", "ask"
    ]
    title: str
    bullets: list[Claim | str]
    status: Literal["complete", "needs_input"]

class ValueProposition(BaseModel):
    pass # To be defined fully by M2

class CompetitorAnalysis(BaseModel):
    pass # To be defined fully by M2

class MarketAnalysis(BaseModel):
    pass # To be defined fully by M2

class BusinessModelAnalysis(BaseModel):
    pass # To be defined fully by M2

class Pitch(BaseModel):
    slides: list[SlideContent]

class PitchCritique(BaseModel):
    issues: list[dict[str, Any]]
    investor_objections: list[str]
    top_priorities: list[str]

class InvestorQuestion(BaseModel):
    id: str
    text: str
    intent: str
    topic: str

class AnswerEvaluation(BaseModel):
    scores: dict[Literal["clarity", "specificity", "evidence", "business_reasoning", "differentiation", "scalability"], int]
    explanation: str
    strengths: list[str]
    weaknesses: list[str]
    recommended_improvement: str
    weakest_criterion: str
    claims_made: list[str]
    follow_up_question: str | None

class ReadinessNarrative(BaseModel):
    pass # To be defined fully by M2

class FeedbackItemDraft(BaseModel):
    pass # To be defined fully by M2
